crates/noirc_evaluator/src/ssa_refactor/ir/dfg.rs

```diff
@@ -465,6 +470,7 @@ impl<'dfg> InsertInstructionResult<'dfg> {
         match self {
             InsertInstructionResult::Results(results) => Cow::Borrowed(results),
             InsertInstructionResult::SimplifiedTo(result) => Cow::Owned(vec![*result]),
+            InsertInstructionResult::SimplifiedToMultiple(results) => Cow::Owned(results.clone()),
```

Comment:
Can we get rid of this clone if we pass self by value into results?

Reply:
Yes we can I did the switch